src/app/pages/collection/pages/collection/nfts/nfts.page.ts

```diff
@@ -52,55 +60,138 @@ export class CollectionNFTsPage implements OnInit, OnChanges {
   spaceFilterOpen = true;
   collectionFilterOpen = true;
   priceFilterOpen = false;
+  sweepCount = 1;
+  originalNfts: any[] = [];
+  private destroy$ = new Subject<void>();
+  availableNftsCount = 0;
+  collection: Collection | null = null;
 
   constructor(
     public filter: FilterService,
     public deviceService: DeviceService,
     public nftApi: NftApi,
+    private collectionApi: CollectionApi,
     public cd: ChangeDetectorRef,
     public filterStorageService: FilterStorageService,
     public cacheService: CacheService,
     public readonly algoliaService: AlgoliaService,
+    public cartService: CartService,
+    private notification: NzNotificationService,
+    private collectionNftStateService: CollectionNftStateService,
   ) {}
 
   public ngOnInit(): void {
-    // Algolia change detection bug fix
-    setInterval(() => this.cd.markForCheck(), 500);
+    console.log('ngOnInit fires');
+    this.collectionNftStateService.availableNftsCount$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((count) => {
+        this.availableNftsCount = count;
+        console.log('[ngOnInit] this.availableNftsCount is set to count: ', count);
+        this.cd.markForCheck();
+      });
+
+    if (this.collectionId) {
+      this.loadCollection(this.collectionId);
+    }
   }
 
-  public ngOnChanges(): void {
-    // TODO comeup with better process.
-    setTimeout(() => {
+  public ngOnChanges(changes: SimpleChanges): void {
+    console.log('ngOnChanges fires');
+    if (changes.collectionId) {
+      this.resetComponentState();
       if (this.collectionId) {
-        this.filterStorageService.marketNftsFilters$.next({
-          ...this.filterStorageService.marketNftsFilters$.value,
-          refinementList: {
-            ...this.filterStorageService.marketNftsFilters$.value.refinementList,
-            collection: [this.collectionId],
-          },
-        });
-
-        this.config = {
-          indexName: COL.NFT,
-          searchClient: this.algoliaService.searchClient,
-          initialUiState: {
-            nft: this.filterStorageService.marketNftsFilters$.value,
-          },
-        };
+        this.loadCollection(this.collectionId);
       }
-    }, 500);
+    }
+  }
+
+  private resetComponentState(): void {
+    console.log('resetComponentState fires');
+    this.availableNftsCount = 0;
+    this.sweepCount = 1;
+    this.cd.markForCheck();
+  }
+
+  private loadCollection(collectionId: string): void {
```

Comment:
I'm not sure exactly how this is used but in case you want to load ALL NFTs from collection you would have to paginate through the whole result. This might take time for some large collections. 

btw: with new APIs we will be to provide aggregate functions.

Reply:
This captures collection of NFTs for purposes of sweeping and then initiates the filter update function.  None of my changes added any additional API queries to fetch NFTs or other index data for this component nor it's parent page.  Once the filter update is finished the altered state triggers the legacy Algolia search-lite client to fetch 20 NFTs for the infinite scroll as normal.